app/src/main/kotlin/com/simplemobiletools/voicerecorder/fragments/TrashFragment.kt

```diff
@@ -0,0 +1,127 @@
+package com.simplemobiletools.voicerecorder.fragments
+
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import android.util.AttributeSet
+import com.simplemobiletools.commons.extensions.*
+import com.simplemobiletools.commons.helpers.ensureBackgroundThread
+import com.simplemobiletools.voicerecorder.R
+import com.simplemobiletools.voicerecorder.activities.SimpleActivity
+import com.simplemobiletools.voicerecorder.adapters.TrashAdapter
+import com.simplemobiletools.voicerecorder.extensions.config
+import com.simplemobiletools.voicerecorder.extensions.getAllRecordings
+import com.simplemobiletools.voicerecorder.interfaces.RefreshRecordingsListener
+import com.simplemobiletools.voicerecorder.models.Events
+import com.simplemobiletools.voicerecorder.models.Recording
+import kotlinx.android.synthetic.main.fragment_trash.view.player_holder
+import kotlinx.android.synthetic.main.fragment_trash.view.recordings_fastscroller
+import kotlinx.android.synthetic.main.fragment_trash.view.recordings_list
+import kotlinx.android.synthetic.main.fragment_trash.view.recordings_placeholder
+import org.greenrobot.eventbus.EventBus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+
+class TrashFragment(context: Context, attributeSet: AttributeSet) : MyViewPagerFragment(context, attributeSet), RefreshRecordingsListener {
+    private var itemsIgnoringSearch = ArrayList<Recording>()
+    private var lastSearchQuery = ""
+    private var bus: EventBus? = null
+    private var prevSavePath = ""
+
+    override fun onResume() {
+        setupColors()
+        if (prevSavePath.isNotEmpty() && context!!.config.saveRecordingsFolder != prevSavePath) {
+            itemsIgnoringSearch = getRecordings()
+            setupAdapter(itemsIgnoringSearch)
+        } else {
+            getRecordingsAdapter()?.updateTextColor(context.getProperTextColor())
+        }
+
+        storePrevPath()
+    }
+
+    override fun onDestroy() {
+        bus?.unregister(this)
+    }
+
+    override fun onAttachedToWindow() {
+        super.onAttachedToWindow()
+
+        bus = EventBus.getDefault()
+        bus!!.register(this)
+        setupColors()
+        itemsIgnoringSearch = getRecordings()
+        setupAdapter(itemsIgnoringSearch)
+        storePrevPath()
+    }
+
+    override fun refreshRecordings() {
+        itemsIgnoringSearch = getRecordings()
+        setupAdapter(itemsIgnoringSearch)
+    }
+
+    override fun playRecording(recording: Recording, playOnPrepared: Boolean) {}
+
+    private fun setupAdapter(recordings: ArrayList<Recording>) {
+        ensureBackgroundThread {
```

Comment:
this bg thread doesnt make much sense, lets remove it from PlayerFragment too

Reply:
Yeah, good catch, I didn't notice that. Both of these just started a background thread and then posted the function back on main thread.